doc/builders/packages/urxvt.xml

```diff
@@ -0,0 +1,90 @@
+<section xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xml:id="sec-urxvt">
+ <title>Urxvt</title>
+
+ <para>
+  Urxvt, also known as rxvt-unicode, is a highly customizable terminal emulator.
+ </para>
+
+ <section xml:id="sec-urxvt-conf">
+
+  <title>Configuring urxvt</title>
+
+  <para>
+   In <literal>nixpkgs</literal>, urxvt is provided by the package
+   <literal>rxvt-unicode</literal>. It can be configured to include your choice
+   of plugins, reducing its closure size from the default configuration which
+   includes all available plugins. To make use of this functionality, use an
+   overlay or directly install an expression that overrides its configuration
+   such as
+   <programlisting>rxvt-unicode.override { configure = { availablePlugins, ... }: {
+    plugins = with availablePlugins; [ perls resize-font vtwheel ];
+  }
+}</programlisting>
+   If the <literal>configure</literal> function returns an attrset without the
+   <literal>plugins</literal> attribute, <literal>availablePlugins</literal>
+   will be used automatically.
+  </para>
+
+  <para>
+   To get a list of all the plugins available, open the Nix REPL and run
+   <programlisting>$ nix repl
+:l &lt;nixpkgs&gt;
+map (p: p.name) pkgs.rxvt-unicode.plugins
+   </programlisting>
+  </para>
+
+  <para>
+    In addition to <literal>plugins</literal> the options
+    <literal>extraDeps</literal> and <literal>perlDeps</literal> can be used
+    to install extra packages. This is a handy way to provide dependencies to
+    your custom perl plugins or install some tools needed by a script.
+    <programlisting>rxvt-unicode.override { configure = { availablePlugins, ... }: {
```

Comment:
Whatever the override scheme will go like eventually, I would like this part to list the required Perl / other dependencies for each plugin. The plugin I packaged once - `bidi`, requires `fribidi`.

Reply:
If a plugin is perl module simply adding it to the `plugins` should just work because its closure already contains all the dependencies.
`perlDeps` is meant to add some perl modules you may need if you are testing/working on a custom plugin (not installed but somewhere in ~)

Reply:
Oh right I'm confused, why are there also `plugins`, `perlDeps`, and `extraDeps` if any given derivation which has a file in it's `$out/lib/urxvt/perl/` will pull all of it's needed dependencies if added to `plugins`? It seems overly complicated now.

BTW 1 more thing I've noticed is that `urxvt_bidi`, the plugin I contributed in the past, is generally a normal perl module, the only thing it has relevant for urxvt is just another file that is installed with `postInstall`, see:

https://github.com/rnhmjoj/nixpkgs/blob/16c22b01b5e706cc0e3b5371ae5409977e618ed8/pkgs/applications/misc/rxvt-unicode-plugins/urxvt-bidi/default.nix#L15-L17

Yet in the old plugins configuration system, it needed to be added also to `perlDeps` in addition to `plugins`... I'm just trying to understand what makes it an exception and perhaps we could overcome this. Perhaps putting the external dependency `fribidi` (which was the original trouble maker) to `propagatedBuildInputs` might help?

Reply:
> Oh right I'm confused, why are there also plugins, perlDeps, and extraDeps if any given derivation which has a file in it's $out/lib/urxvt/perl/ will pull all of it's needed dependencies if added to plugins? It seems overly complicated now.

I've rephrased the explanation I give in the manual and added two examples. I hope it's clear now. The idea is basically:

* `plugins`: to add urxvt plugins
* `extraDeps`: to add any kind of package that it's not specifically a plugin to the wrapper
* `perlDeps`: to add Perl packages/modules so that you can use them in your plugins, which are not packaged in nixpkgs but are in `~/.urxvt/ext`

> BTW 1 more thing I've noticed is that urxvt_bidi

What's special about `urxvt_bidi` is that it's a Perl package (you built it with `buildPerlPackage`) and that it needs to be added to the `PERL5LIB` path. Other plugins are simple Perl scripts that urxvt looks for in the `URXVT_PERL_LIB` path.